tiledb/sm/query/readers/sparse_index_reader_base.cc

```diff
@@ -240,8 +240,17 @@ std::vector<uint64_t> SparseIndexReaderBase::tile_offset_sizes() {
           num += deletes_consolidation_no_purge_;
         }
 
+        // Other than the offsets themselves, there is also memory used for the
+        // initialization of the vectors that hold them. This initialization
+        // takes place in fragment_metadata.cc::load_footer()
+        unsigned num_vectors = schema->attribute_num() + 1 +
+                               fragment->has_timestamps() +
+                               fragment->has_delete_meta() * 2;
+        num_vectors += (fragment->version() >= 5) ? schema->dim_num() : 0;
```

Comment:
Add a comment on this line as to what it does... For fragments up to version 4, we use zipped coordinates.

Reply:
Done